src/lib.rs

```diff
@@ -381,25 +382,24 @@ where
     }
 }
 
-impl<T, P> AtomicPtr<T, domain::Global, P> {
+impl<T, F, P> AtomicPtr<T, F, P> {
     /// Loads the value from the stored pointer and guards it using the given hazard pointer.
     ///
     /// The guard ensures that the loaded `T` will remain valid for as long as you hold a reference
     /// to it.
     ///
-    /// Note that this method is _only_ available when using [`Domain<Global>`](domain::Global),
-    /// since it is a singleton, and thus is guaranteed to fulfill the safety requirement of
-    /// [`AtomicPtr::load`]. For all other domains, use [`AtomicPtr::load`].
-    pub fn safe_load<'hp>(
-        &'_ self,
-        hp: &'hp mut HazardPointer<'static, domain::Global>,
-    ) -> Option<&'hp T>
+    /// This method is a safe alternative to [`AtomicPtr::load`] available to
+    /// [`Domain::Singleton`](singleton families) since it is a singleton, and thus
+    /// is guaranteed to fulfill the safety requirement of [`AtomicPtr::load`]
+    pub fn safe_load<'hp, 'd>(&'_ self, hp: &'hp mut HazardPointer<'d, F>) -> Option<&'hp T>
     where
         T: 'hp,
+        F: 'static,
+        F: Singleton,
```

Comment:
I think I'd prefer this bound to be on the `impl` block rather than just the method. Makes the docs very slightly easier to parse.